ovirt_imageio/client/_ui.py

```diff
@@ -15,87 +15,81 @@
 
 class ProgressBar:
 
-    def __init__(self, size=0, output=sys.stdout, step=0.1, width=79,
+    def __init__(self, size=None, output=sys.stdout, step=None, width=79,
                  now=time.monotonic):
         """
-        Argumnets:
+        Arguments:
             size (int): total number of bytes. If size is unknown when
                 creating, progress value is not displayed. The size can be set
                 later to enable progress display.
             output (fileobj): file to write progress to (default sys.stdout).
-            step (float): mininum progress update interval in seconds (default
-                0.1 seconds).
+            step (float): unused, kept for backward compatibility. The progress
+                is updated in 1 percent steps.
             width (int): width of progress bar in characters (default 79)
             now (callable): callable returning current time for testing.
         """
         self.size = size
-        self.output = output
-        self.step = step
+        self._output = output
         # TODO: use current terminal width instead.
-        self.width = width
-        self.now = now
-        self.lock = threading.Lock()
-        self.start = self.now()
-        self.next = 0
-        self.done = 0
+        self._width = width
+        self._now = now
+        self._lock = threading.Lock()
+        self._start = self._now()
+        # Number of bytes transferred.
+        self._done = 0
+        # Value in percent. We start with -1 so the first update will show 0%.
```

Comment:
But now we call _draw immediately, not update any more. So we will draw first with self._value -1, at least if size was specified. Is that what we want?

Reply:
This does not happen in current code since we never set size when creating the progress bar, but it is a bug. We also miss a test for setting size upfront.

Reply:
Fixed in master now.